### ISSUE TITLE
CayleyDigraph can return wrong answer

### ISSUE THREAD

**Issue by wilfwilson** [digraphs/Digraphs#366]
This was discovered by @jdebeule (see #363). Thanks for spotting this. For example:
```
gap> G := DihedralGroup(8);;
gap> gens := GeneratorsOfGroup(G);;
gap> elts := AsList(G);;
gap> D := CayleyDigraph(G, gens);;
gap> Filtered(DigraphEdges(D), edge ->
> not ForAny(gens, x -> elts[edge[1]] * x = elts[edge[2]]));
[ [ 2, 8 ], [ 3, 5 ], [ 5, 3 ], [ 5, 2 ], [ 6, 5 ], [ 7, 8 ], [ 8, 7 ], [ 8, 6 ] ]
```
This is incorrect: the pair `[i,j]` is an edge of the Cayley digraph if and only if the `i`th element multiplied by some generator is the `j`th element. So the final command should return an empty list.

**Comment by james-d-mitchell** [comment]
Thanks for the report @wilfwilson, am I correct in thinking that this can be resolved by replacing the use of `OnRight` in line 110 of `grape.gi` with `OnLeftInverse`?

**Comment by jdebeule** [comment]
Dear James, dear Wilf, yes, it is sufficient to replace the action function OnRight by OnLeftInverse. I am not sure whether I did the pull request correctly, but exactly this correction is in the pull request, together with some other things.

**Comment by wilfwilson** [comment]
Yeah @jdebeule's PR makes this change, I created the issue to help us keep track of the bug. And I'll happily look at PR #378 and merge the fix as appropriate (even though Jan's PR has it too) because it's good to have the bug fixed, and I've not got around to reviewing Jan's slightly larger PR yet.

**Comment by james-d-mitchell** [comment]
Resolved in v1.4.0
